autozone-poc/src/utlis/formComponents/InputWithLabel.js

```diff
@@ -21,14 +32,51 @@ export default function TextInput({
             className="form-control"
             id={inputId}
             required={isRequired}
+            onBlur = {textValidationHandler}
             defaultValue=""
             {...etc}
           />
+          <p className="formErrorMsg"></p>
+        </>
+      );
+      break;
+
+      case "inputWithoutLabel":
+      function inputValidationHandler(){
+        let inputData = $(`#${inputId}`).val();
+        if(isRequired && inputData === ''){
+          $(`#${inputId} + .formErrorMsg`).text("Please provide some Input");
+          $(`#${inputId} + .formErrorMsg`).show();
+        }else{
+          $(`#${inputId} + .formErrorMsg`).hide();
+        }
+      }
+      inputLabel = (
+        <>
+          <input
+            type="text"
+            className="form-control"
+            id={inputId}
+            required={isRequired}
+            onBlur = {inputValidationHandler}
+            defaultValue=""
+            {...etc}
+          />
+          <p className="formErrorMsg"></p>
         </>
       );
       break;
 
     case "email":
+      function emailValidationHandler(){
+        let inputData = $(`#${inputId}`).val();
+        if(isRequired && (inputData === '' || inputData.indexOf('@') === -1 || inputData.indexOf('.') === -1)){
+          $(`#${inputId} + .formErrorMsg`).text("Please provide proper email address");
+          $(`#${inputId} + .formErrorMsg`).show();
+        }else{
+          $(`#${inputId} + .formErrorMsg`).hide();
+        }
+      }
       inputLabel = (
         <>
```

Comment:
I don't see there is an option to handle a non-required field. Kindly add that option in all fields as well as prop types also.

Reply:
@gokulwebdeveloper  As discussed, non-required field will not have any validation functionality.

Reply:
@gokulwebdeveloper : please approve.